cdci_data_analysis/analysis/renku_helper.py

```diff
@@ -65,17 +62,26 @@ def push_api_code(api_code,
         raise RequestNotUnderstood(error_message)
     finally:
         logger.info("==> removing repository folder, since it is no longer necessary")
-        remove_repository(renku_repository_url, repository_folder_path)
-    # TODO to actually return the renkulab url of the newly created branch
+        remove_repository(repo, renku_repository_url)
+
     return renku_session_url
 
 
-def generate_renku_session_url(repo, renku_project_url, renku_gitlab_user_name, branch_name):
-    # original_url = repo.remotes.origin.url
-    # TODO: project url could be derived from renku projects base path and original_url. the config values should be renamed then    
+def generate_renku_session_url(repo, renku_base_project_url, branch_name):
+    original_url = repo.remotes.origin.url
+    repo_path = get_repo_path(original_url)
+    renku_project_url = f'{renku_base_project_url}/{repo_path}'
     return f"{renku_project_url}/sessions/new?autostart=1&branch={branch_name}"
     
 
+def get_repo_path(repository_url):
+    if repository_url.endswith('.git'):
```

Comment:
this might be too delicate, and break in unexpected cases. Is there not a way to parse remotes in git-python? Else two regexes would be better, and exception if the format does not match.

Reply:
this seems to be a good library for this specific purpose

[https://pypi.org/project/giturlparse/](https://pypi.org/project/giturlparse/)